### PR TITLE
GRIF-165: fix -a flag to return all results

### PR DESCRIPTION
**Pull request by JimFuller-RedHat**
The following should include upstream components (by default)
```
> griffon service products-contain-component openshift/console --search-all -a  
```
To disable this behaviour use **--no-upstream-affects** flag
```
> griffon service products-contain-component openshift/console --search-all -a  --no-upstream-affects
```

### CODE REVIEW

File: griffon/output.py [hunk 1]
```diff
@@ -349,11 +346,13 @@ def text_output_products_contain_component(
...
 
             flaw_mode = ctx.params["flaw_mode"]
-            flaw_operation = "dry_run"
+            flaw_operation = "new"
             if flaw_mode == "add":
                 flaw_operation = "new"
             if flaw_mode == "update":
                 flaw_operation = "update"
+            if flaw_mode == "dry_run":
+                flaw_operation = "dry_run"
```
Comment (JakubFrejlach) [inline griffon/output.py:366]: Nitpick: Maybe some dict with `flaw_mode` to `flaw_operation` mapping would be clearer:
```python
flaw_mode_to_operation = dict(...)
flaw_operation = flaw_mode_to_operation.get(flaw_mode, "new")
```

File: griffon/output.py [hunk 2]
```diff
@@ -216,6 +216,14 @@ def generate_normalised_results(
...
                                 if item.get("type") == output_type_filter:
                                     normalised_results.append(c)
+                            else:
+                                logger.debug(f'.griffon blocked component {item["name"]}')
+                        else:
+                            logger.debug(f'prod def blocked component {item["name"]}')
+                    else:
+                        logger.debug(f'.griffonrc product streams blocked component {item["name"]}')
+                else:
+                    logger.debug(f'product stream not active {ps}')
```
Comment (JakubFrejlach) [inline griffon/output.py]: Nitpick: maybe in the future, we would like to restructure this for better readability by flipping the conditions to something like this:
```python
if condition1:
    logger.debug(...)
    continue

if condition2:
    logger.debug(...)
    continue
    
...

do_something_with_data()
```

File: griffon/exceptions.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+"""
+
+
+"""
```
Comment (JakubFrejlach) [inline griffon/exceptions.py:4]: No description?

**Review by JakubFrejlach — APPROVED** [review]
Couple of suggestions, but otherwise LGTM.

File: griffon/exceptions.py (hunk 3 above)
Comment (JimFuller-RedHat) [reply griffon/exceptions.py]: added

File: griffon/output.py (hunk 2 above)
Comment (JimFuller-RedHat) [reply griffon/output.py]: removed debug statements (only needed them temporarily)

File: griffon/output.py (hunk 1 above)
Comment (JimFuller-RedHat) [reply griffon/output.py:358]: all this affect machinery is tactical ... going to leave asis for now.

**Pull request merged**